test/grouping.jl

```diff
@@ -21,4 +21,8 @@ d = DataFrame(n = 1:20, x = [3, 3, 3, 3, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2, 2, 3,
 g = groupby(d, :x, sort=true)
 @test @based_on(g, nsum = sum(:n))[:nsum] == [99, 84, 27]
 
+d = DataFrame(a = [1,1,1,2,2], b = [1,2,3,missing,missing])
```

Comment:
If you expect a particular behavior for `CategoricalArray`, it should be tested (and a dependency on it be added only for testing).